core/trino-main/src/main/java/io/trino/execution/buffer/SpoolingExchangeOutputBuffer.java

```diff
@@ -159,11 +166,13 @@ public void enqueue(int partition, List<Slice> pages)
             return;
         }
 
+        ExchangeSink sink = exchangeSink;
```

Comment:
why not to inline? `exchangeSink` is not volatile and this seems to be used from single thread

Reply:
`enqueue` and `abort` methods can be called from different threads.

Reply:
This part was not obvious to me too. Maybe it is worth documenting in the `OutputBuffer` thread safety semantics. What methods should be run in thread-safe manner with external synchronization. And what methods can be run from different threads.

Reply:
Generally an `OutputBuffer` implementation is expected to be thread safe. There's no guarantees on what methods can be called from what threads.

Reply:
But then what you wrote above would not be true (that `abort` can be run from different thread than `enqueue`).